src/AstLiteral.h

```diff
@@ -49,6 +49,14 @@ class AstAtom : public AstAtomLiteral {
 public:
     AstAtom(AstRelationIdentifier name = AstRelationIdentifier()) : name(std::move(name)) {}
 
+    // construct AstAtom with given atom and replace the arguments with args
+    AstAtom(const AstAtom& atom, std::vector<AstArgument*> args) : name(atom.getName()) {
```

Comment:
This should take a srcLocation, name, and vector of unique pointers, or to be consistent with the rest of the Ast, name + vector then call setSrcLoc().

If you need to construct it from an existing atom for MaterialiseAggregationQueriesTransformer then you could add a helper method/lambda there.